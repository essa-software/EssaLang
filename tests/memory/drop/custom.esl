struct TestDrop {
    id: u32;

    func __drop__(this) {
        print("Drop! {}\n", this.id);
    }
}

func main() {
    print("Construct TestDrop 1\n");
    mut td1: TestDrop = TestDrop { id: 1 };

    print("Construct TestDrop 2\n");
    mut td2: TestDrop = TestDrop { id: 2 };

    print("Before drop\n");
}
