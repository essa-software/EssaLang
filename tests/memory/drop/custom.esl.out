Construct TestDrop 1
Construct TestDrop 2
Before drop
Drop! 2
Drop! 1
